app/src/main/java/com/amaze/filemanager/filesystem/files/GenericCopyUtil.java

```diff
@@ -315,11 +313,22 @@ else if (outChannel != null) {
         if (bufferedOutputStream != null) copyFile(inChannel, bufferedOutputStream);
         else if (outChannel != null) copyFile(inChannel, outChannel);
       }
+      // till now file might not be written to disk yet, but to an OS buffer.
+      if (bufferedOutputStream != null) {
```

Comment:
You might never get here if an exception ocurrs.

Reply:
I know

Reply:
This will probably not be accepted unless you fix this.

Reply:
fix what? nothing is broken there

Reply:
if the code crashes stream is not closed.

Reply:
The stream is closed no matter exception or not